app/src/editor/canvas/components/CodeEditorWindow.jsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react'
+import AceEditor from 'react-ace'
+
+import 'brace/mode/java'
+import 'brace/theme/textmate'
+
+import DraggableCanvasWindow from './DraggableCanvasWindow'
+
+import styles from './CodeEditorWindow.pcss'
+
+class CodeEditorWindow extends React.Component {
+    state = {
+        code: this.props.code,
```

Comment:
Should probably support possibility of different code arriving after mount, but obviously don't replace the code if the user is currently editing. I believe there's logic for this in the editor TextInput component.